static/app/components/profiling/ThreadSelector.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import Fuse from 'fuse.js';
+
+import AutoComplete from 'sentry/components/autoComplete';
+import Button from 'sentry/components/button';
+import Input from 'sentry/components/forms/controls/input';
+import {IconCheckmark} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {ProfileGroup} from 'sentry/utils/profiling/profile/importProfile';
+import {Profile} from 'sentry/utils/profiling/profile/profile';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+const sortProfiles = (profiles: ReadonlyArray<Profile>): ProfileGroup['profiles'] => {
+  return [...profiles].sort((a, b) => {
+    if (!b.duration) {
+      return -1;
+    }
+    if (!a.duration) {
+      return 1;
+    }
+
+    if (a.name.startsWith('(tid') && b.name.startsWith('(tid')) {
+      return -1;
+    }
+    if (a.name.startsWith('(tid')) {
+      return -1;
+    }
+    if (b.name.startsWith('(tid')) {
+      return -1;
+    }
+    if (a.name.includes('main')) {
+      return -1;
+    }
+    if (b.name.includes('main')) {
+      return 1;
+    }
+    return a.name > b.name ? -1 : 1;
+  });
+};
+
+interface ThreadSelectorProps {
+  activeProfileIndex: ProfileGroup['activeProfileIndex'];
+  onProfileIndexChange: (index: number) => void;
+  profileGroup: ProfileGroup;
+}
+
+function ThreadMenuSelector(props: ThreadSelectorProps): React.ReactElement | null {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [open, setOpen] = React.useState<boolean>(false);
+
+  const handleSelectItem = React.useCallback(
+    (p: Fuse.FuseResultWithMatches<Profile & {index: number}>) => {
+      const index = props.profileGroup.profiles.findIndex(
+        profile => profile.name === p.item.name
```

Comment:
Are the profiles in each profile group guaranteed to have unique names?

Reply:
Yes, the profile name is a mix of thread name + thread id, so they dont get mixed.